server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -195,6 +213,64 @@ public Status getStatus() {
         return null;
     }
 
+    /**
+     * Returns total resource consumption of the task
+     */
+    public Map<String, List<TaskCompleteResourceInfo>> getResourceStats() {
+        return resourceStats;
+    }
+
+    public Map<String, Long> getTotalResourceStats() {
+        return new HashMap<String, Long>() {
+            {
+                for (TaskStats taskStats : TaskStats.values()) {
+                    put(taskStats.toString(), getTotalResourceUtilization(taskStats));
+                }
+            }
+        };
+    }
+
+    /**
+     * Returns total resource consumption for a specific task stats.
+     */
+    public long getTotalResourceUtilization(TaskStats taskStats) {
+        long totalResourceConsumption = 0L;
+        for (List<TaskCompleteResourceInfo> resourceInfoList : resourceStats.values()) {
+            for (TaskCompleteResourceInfo resourceInfo : resourceInfoList) {
+                for (Map.Entry<TaskStatsType, TaskResourceInfo> entry : resourceInfo.getResourceInfo().entrySet()) {
+                    if (!entry.getKey().isOnlyForAnalysis()) {
+                        totalResourceConsumption += entry.getValue().getStatsInfo().get(taskStats).getTotalValue();
+                    }
+                }
+            }
+        }
+        return totalResourceConsumption;
+    }
+
+    /**
+     * Add or update thread's resource consumption stats
+     * @param threadId: threadId
```

Comment:
Nitpick, but the javadoc style does not use colons after the parameter name

Reply:
Will fix this.